source/common/network/listen_socket_impl.h

```diff
@@ -153,6 +153,25 @@ class UdsListenSocket : public ListenSocketImpl {
   Socket::Type socketType() const override { return Socket::Type::Stream; }
 };
 
+// This socket type adapts the ListenerComponentFactory.
+class InternalListenSocket : public ListenSocketImpl {
+public:
+  InternalListenSocket(const Address::InstanceConstSharedPtr& address)
+      : ListenSocketImpl(/* io_handle= */ nullptr, address) {}
+  Socket::Type socketType() const override { return Socket::Type::Stream; }
+
+  // InternalListenSocket cannot be duplicated.
+  SocketPtr duplicate() override {
+    return std::make_unique<InternalListenSocket>(connectionInfoProvider().localAddress());
+  }
+
+  void close() override { ASSERT(io_handle_ == nullptr); }
+  bool isOpen() const override {
+    ASSERT(io_handle_ == nullptr);
+    return false;
```

Comment:
Why is this false?

Reply:
The close does nothing. So I assume either blindly close or close-while-isOpen works as before. I don't see what would check a listen socket isopen.

A flag is_open can be added later if we do need to maintain the isOpen state, but I think we are not there yet.